compiler/AST/ImportStmt.cpp

```diff
@@ -294,6 +316,82 @@ bool ImportStmt::checkValid(Expr* expr) const {
   return retval;
 }
 
+/************************************* | **************************************
+*                                                                             *
+* Verifies that all the symbols in the list for unqualified access of import  *
+* statements refer to symbols that are visible from that module.              *
+*                                                                             *
+************************************** | *************************************/
+void ImportStmt::validateList() {
+  noRepeats();
+
+  validateUnqualified();
+  validateRenamed();
+}
+
+void ImportStmt::noRepeats() const {
+  std::vector<const char*>::const_iterator           it1;
+
+  for (it1 = unqualified.begin(); it1 != unqualified.end(); ++it1) {
+    std::vector<const char*>::const_iterator next = it1;
+    std::map<const char*, const char*>::const_iterator rit;
+
+    for (++next; next != unqualified.end(); ++next) {
+      // Check rest of named for the same name
+      if (strcmp(*it1, *next) == 0) {
+        USR_WARN(this, "identifier '%s' is repeated", *it1);
+      }
+    }
+
+    for (rit = renamed.begin(); rit != renamed.end(); ++rit) {
+      if (strcmp(*it1, rit->second) == 0) {
+        // This identifier is also used as the old name for a renaming.
+        // Probably a mistake on the user's part, but not a catastrophic one
+        USR_WARN(this, "identifier '%s' is repeated", *it1);
+      }
+
+      if (strcmp(*it1, rit->first) == 0) {
+        // The user attempted to rename a symbol to a name that was already
+        // in the 'only' list.  This causes a naming conflict.
+        USR_FATAL_CONT(this, "symbol '%s' multiply defined", *it1);
+      }
+    }
+  }
+
+  noRepeatsInRenamed();
+}
+
+void ImportStmt::validateUnqualified() {
+    BaseAST*            scopeToUse = getSearchScope();
+  const ResolveScope* scope      = ResolveScope::getScopeFor(scopeToUse);
+
+  for_vector(const char, name, unqualified) {
+    if (name[0] != '\0') {
+      std::vector<Symbol*> symbols;
+
+      scope->getFields(name, symbols);
```

Comment:
lazy question: this gets the function symbols, too, right?

Reply:
Having gone through the PR as a whole, I haven't seen any particular test that exercises these checks, so I still have the same question

Reply:
I believe it will, but I haven't tested it yet.  However, UseStmts use roughly the same code and I believe I test that for functions?